src/Model/ModelManager.php

```diff
@@ -500,17 +499,16 @@ public function addIdentifiersToQuery($class, ProxyQueryInterface $query, array
 
         $fieldNames = $this->getIdentifierFieldNames($class);
         $qb = $query->getQueryBuilder();
+        $classMetadata = $this->getMetadata($class);
 
-        $prefix = uniqid();
+        $prefix = $this->generateIdentifierPrefix();
```

Comment:
to have an ability to override it in tests, otherwise, it's impossible to predict results with hardcoded `unique`

Reply:
We shouldn't add method for this.
You could check the name by using a regex instead.

Reply:
One good practice from DDD - avoid having dependencies on I/O outside of the Infrastructure layer, so I tend to remove such dependencies into a separate method/class so that it could be replaced later without affecting main functionality and tests. What do you think about it?

Reply:
I like the idea 💡

Reply:
@oleg-andreyev In Sonata 4.0, the ModelManager will be final.
- So a protected method is useless, it's basically a private one.
- And creating a private method for calling `uniqid()` seems useless to me.